src/main/java/com/aws/greengrass/lifecyclemanager/Lifecycle.java

```diff
@@ -680,14 +680,13 @@ private void handleCurrentStateErrored(Optional<State> desiredState, State prevS
         }
 
         switch (prevState) {
+            // For new, we can interrupt it and cancel the installation when it is requested to shutdown nucleus
             // For both starting and running, make sure we stop first before retrying
+            case NEW:
             case STARTING:
             case RUNNING:
                 internalReportState(State.STOPPING);
                 break;
-            case NEW: // error in installing.
-                internalReportState(State.NEW);
-                break;
```

Comment:
Install should not go through the stopping phase. Stop only runs after `run` or `startup`. Install should be interruptable and using the `close` method of `Exec`.